packages/query/src/graphql/plugins/PgOrderByUnique.ts

```diff
@@ -0,0 +1,177 @@
+                  specs: [],
+                },
+              },
+            },
+          },
+          {
+            __origin: `Adding connection "orderBy" argument for ${describePgEntity(
+              table
+            )}. You can rename the table's GraphQL type via a 'Smart Comment':\n\n  ${sqlCommentByAddingTags(table, {
+              name: 'newNameHere',
+            })}`,
+            pgIntrospection: table,
+            isPgRowSortEnum: true,
+          }
+        );
+      });
+
+      return _;
+    },
+    ['PgConnectionArgOrderBy']
+  );
+
+  builder.hook(
+    'GraphQLObjectType:fields:field:args',
+    (args, build, context) => {
+      const {
+        extend,
+        getTypeByName,
+        graphql: {GraphQLList, GraphQLNonNull},
+        inflection,
+        pgGetGqlTypeByTypeIdAndModifier,
+        pgOmit: omit,
+        pgSql: sql,
+      } = build;
+      const {
+        Self,
+        addArgDataGenerator,
+        scope: {
+          fieldName,
+          isPgFieldConnection,
+          isPgFieldSimpleCollection,
+          pgFieldIntrospection,
+          pgFieldIntrospectionTable,
+        },
+      } = context;
+
+      if (!isPgFieldConnection && !isPgFieldSimpleCollection) {
+        return args;
+      }
+
+      const proc = pgFieldIntrospection.kind === 'procedure' ? pgFieldIntrospection : null;
+      const table: PgClass | null =
+        pgFieldIntrospection.kind === 'class' ? pgFieldIntrospection : proc ? pgFieldIntrospectionTable : null;
+
+      if (!table || !table.namespace || !table.isSelectable || omit(table, 'order')) {
+        return args;
+      }
+
+      if (proc) {
+        if (!proc.tags.sortable) {
+          return args;
+        }
+      }
+
+      const TableType = pgGetGqlTypeByTypeIdAndModifier(table.type.id, null);
+      const tableTypeName = TableType.name;
+      const TableOrderByType = getTypeByName(inflection.orderByType(tableTypeName)) as GraphQLEnumType;
+
+      const cursorPrefixFromOrderBy = (orderBy: any) => {
+        if (orderBy) {
+          const cursorPrefixes = [];
+
+          for (let itemIndex = 0, itemCount = orderBy.length; itemIndex < itemCount; itemIndex++) {
+            const item = orderBy[itemIndex];
+
+            if (item.alias) {
+              cursorPrefixes.push(sql.literal(item.alias));
+            }
+          }
+
+          if (cursorPrefixes.length > 0) {
+            return cursorPrefixes;
+          }
+        }
+
+        return null;
+      };
+
+      addArgDataGenerator(function connectionOrderBy({orderBy: rawOrderBy}: any) {
+        const orderBy = rawOrderBy ? (Array.isArray(rawOrderBy) ? rawOrderBy : [rawOrderBy]) : null;
+        return {
+          pgCursorPrefix: cursorPrefixFromOrderBy(orderBy),
+          pgQuery: (queryBuilder) => {
+            if (orderBy !== null) {
+              orderBy.forEach((item) => {
+                console.log('item: ', item);
+                const {specs, unique} = item;
```

Comment:
How does `unique` get set? It seem the functionality already exists and the query could be modified to improve performance already

Reply:
After reading the docs this only seems to be set if you specify a specific order by https://www.graphile.org/postgraphile/make-add-pg-table-order-by-plugin/#orderbyascdesc